bitcoin/tests/psbt.rs

```diff
@@ -439,7 +437,7 @@ fn finalize_psbt(mut psbt: Psbt) -> Psbt {
     // Input 1: SegWit UTXO
 
     let script_sig = script::Builder::new()
-        .push_slice(&psbt.inputs[1].redeem_script.clone().unwrap().serialize())
+        .push_slice(&psbt.inputs[1].redeem_script.clone().unwrap().as_bytes())
```

Comment:
I know this is in tests but it still takes reviewer time to realize that. I'd prefer `redeem_script.as_ref().unwrap().as_bytes()`

Reply:
May you share more about this preference? Is this just to avoid the clone()?

Reply:
I'll take a guess at what he means :) The `clone` causes reviewers to stop and think "why is there a clone", then they have to think more and see that its in test code so its ok to clone. Based on the idea that we should always optimize for reviewers (because review time is always in short supply) it would be better if there was no `clone`.

How'd I go @Kixunil?

Reply:
And ironically that long response probably wastes reviewers time :)

Reply:
and now I won't waste it in the future 😄

Reply:
Yes, what @tcharding said and this is amplified by the code pretty obviously not needing it since all other parts work on references.